### PR TITLE
cam6_2_046: Test cleanup

### LINKED ISSUE #174: Removal of CAM standalone testing
Remove the CAM standalone testing from test_driver prior to the release

### LINKED ISSUE #193: test_driver.sh can find wrong CIME
If you are running `test_driver.sh` from a standalone CAM clone _and_ you happen to have a CIME clone at the same directory level as the CAM clone, `test_driver.sh` will try to use the CIME clone instead of the CIME version under CAM.

### LINKED ISSUE #198: DAE test is broken
The DAE test (DAE.f45_f45_mg37.FHS94.izumi_nag.cam-dae) has been broken since cam6_1_033, however, it was not being tested until now (when I moved the test from Hobart to Izumi).

### LINKED ISSUE #199: CAM standalone tests no longer needed
Now that all CAM tests are using the CIME infrastructure, the CAM standalone tests and associated infrastructure are no longer needed.

### PR DESCRIPTION
**Pull request by gold2718**
Remove CAM standalone tests
Fixes #174 
Fix DAE test is broken
Fixes #198 
Fix CAM root and CIME finding in test_driver.sh
Fixed #193

In addition, the CAM build system has been setting namelist variables in other
components which interferes with the CIME case control system's ability
to keep track of changes. These settings have been removed.

Note, this means that `<start_ymd>` is ignored, e.g., in use-case files.

### CODE REVIEW

File: cime_config/buildnml [hunk 1]
```diff
@@ -39,6 +39,8 @@ def buildnml(case, caseroot, compname):
     CAM_NML_USE_CASE    = case.get_value("CAM_NML_USE_CASE")
     DEBUG               = case.get_value("DEBUG")
     NINST_ATM           = case.get_value("NINST_ATM")
+    NTASKS              = case.get_value("NTASKS_PER_INST_ATM")
+    DOCN_MODE           = case.get_value("DOCN_MODE")
```
Comment (cacraigucar) [inline cime_config/buildnml:43]: Why add this, as I don't see it being used anywhere?

File: doc/ChangeLog [hunk 2]
```diff
@@ -1,5 +1,371 @@
...
+ - Pass new argument to cam_ctrl_init
+M src/control/cam_control_mod.F90
+ - Added new module variable, post_assim, to indicate a resume condition
+ - Set post_assim in cam_ctrl_init
+ - Use new variable to print appropriate initialization message
+M src/control/cam_history_support.F90
+ - Add missing field item in field_copy
+M src/cpl/mct/atm_comp_mct.F90
+ - Pass 'dart_mode' to cam_init as post_assim_in
+test/system/test_driver.sh
```
Comment (cacraigucar) [inline doc/ChangeLog:327]: Put "M" in front for consistency and ease of browsing

File: doc/ChangeLog [hunk 3]
```diff
@@ -1,5 +1,371 @@
...
+test/system/tests_pretag_cheyenne
+test/system/tests_pretag_hobart_nag
+test/system/tests_pretag_hobart_pgi
+test/system/tests_pretag_izumi_nag
+test/system/tests_pretag_izumi_pgi
+test/system/tests_pretag_leehill
+test/system/tests_waccm_hybrid
+test/system/tests_waccm_mpi
+test/system/tests_waccmx_hybrid
+test/system/tests_waccmx_mpi
```
Comment (cacraigucar) [inline doc/ChangeLog:252]: Need a summary comment of why these files are removed

File: doc/ChangeLog (hunk 2 above)
Comment (gold2718) [reply doc/ChangeLog:327]: I looked and there is no consistency in this usage. If you really feel that a particular format is required, change the template to indicate this.

File: doc/ChangeLog (hunk 3 above)
Comment (gold2718) [reply doc/ChangeLog:252]: I just followed the directions (which do not say anything about a comment). Does the template need to be updated?

File: doc/ChangeLog (hunk 2 above)
Comment (cacraigucar) [reply doc/ChangeLog:327]: I'm just saying consistency with your other modified files above - they all have "M".  My quick scan through visually tripped on this line as it was "different"

File: cime_config/buildnml (hunk 1 above)
Comment (gold2718) [reply cime_config/buildnml:43]: NTASKS is used on line 155
I had a use for DOCN_MODE but I guess it went away with a different refactor. I'll take it out, thanks

File: doc/ChangeLog (hunk 3 above)
Comment (cacraigucar) [reply doc/ChangeLog:252]: I believe we've always in the past commented on any files eliminated at least with a summary of why they were removed.  See cam6_2_000 where you commented on an eliminated file

File: cime_config/buildnml (hunk 1 above)
Comment (cacraigucar) [reply cime_config/buildnml:43]: Sorry - the highlighter grabbed one more line than it should have.  I was only asking about DOCN_MODE

**Comment by gold2718** [comment]
>   Do we want any sort of git sanity checks now that TGIT.sh is gone?
> 
> It looks like TR8.sh won't be called anymore. Were there plans to try and call it from somewhere else (at least until we create a Github Action for it)?

I ran these by hand (resurrected TGIT), perhaps we just add them to test_driver.sh in the future (see ChangeLog).

**Pull request merged**